datalad/distributed/ora_remote.py

```diff
@@ -630,32 +630,48 @@ class HTTPRemoteIO(object):
     # NOTE: For now read-only. Not sure yet whether an IO class is the right
     # approach.
 
-    def __init__(self, ria_url, dsid, buffer_size=DEFAULT_BUFFER_SIZE):
-        assert ria_url.startswith("ria+http")
-        self.base_url = ria_url[4:]
-        if self.base_url[-1] == '/':
-            self.base_url = self.base_url[:-1]
+    def __init__(self, url, buffer_size=DEFAULT_BUFFER_SIZE):
+        if not url.startswith("http"):
+            raise RIARemoteError("Expected HTTP URL, but got {}".format(url))
+
+        self.store_url = url
+        if self.store_url[-1] == '/':
+            self.store_url = self.store_url[:-1]
```

Comment:
Unrelated to this PR, but if downstream code requires a URL to not have a trailing slash, this better be an unconditional `.rstrip('/')`.